# define parameters for calibration and identification process
# robot: tiago
---
calibration:
  calib_level: full_params # full_params /joint_offset
  non_geom: False
  base_frame: Base # start_frame of kinematic chain
  tool_frame: end_effector # end_frame of kinematic chain
  base_to_ref_frame:  # camera frame
  ref_frame:  # parent joint frame of camera frame
  markers:
    - ref_joint: q_3
      measure:  [True, False, False, False, False, False]
    # - ref_joint: q_3
    #   measure: [False, False, True, False, False, False]
  free_flyer: False
  camera_pose: [0.3, 0.0, 0.0, 0, 0.0, 1.57]
  tip_pose: [0, 0, 0, 0, 0, 0]
  coeff_regularize: 0.1
  outlier_eps: 0.15 #meter
  # data_file: data/eye_hand_calibration_recorded_data_palhey5_17_test1.csv
  data_file: data/10_points_x.csv
  sample_configs_file: data/optimal_configs/eye_hand_calibration_tiago_hey5_100.yaml
  nb_sample: 100
identification:
  robot_params:
    - q_lim_def: 1.57
      fv : [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
      fs : [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
      Ia : [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
      offset : [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
      Iam6 : 0
      fvm6 : 0
      fsm6 : 0
      N : [32.0, 32.0, 45.0, -48.0, 45.0, 32.0]
      ratio_essential : 30.0
      dq_lim_def : [0.05009095, 0.05009095, 0.07504916, 0.0715585,  0.05585054, 0.12217305]
      ddq_lim_def : 20.0
      tau_lim_def : 4.0
  problem_params:
    - is_external_wrench : False
      is_joint_torques : True
      force_torque : None
      external_wrench_offsets : False
      has_friction : True
      has_joint_offset : True
      has_actuator_inertia : True
      is_static_regressor : True
      is_inertia_regressor : True
      has_coupled_wrist: True
      embedded_forces : False
  processing_params:
    - cut_off_frequency_butterworth: 100.0
      ts : 0.0002
  tls_params:
    - mass_load : 0.0
      which_body_loaded : 0.0
      sync_joint_motion : False
...
